packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -121,15 +121,33 @@ abstract class RenderSliverBoxChildManager {
   /// true without making any assertions.
   bool debugAssertChildListLocked() => true;
 }
+/// Parent data structure used by [RenderSliverWithKeepAliveMixin].
+abstract class KeepAliveParentDataMixin extends ParentData {
+  // This class is intended to be used as a mixin, and should not be
+  // extended directly.
+  factory KeepAliveParentDataMixin._() => null;
+
+  /// Whether to keep the child alive even when it is no longer visible.
+  bool keepAlive = false;
+}
```

Comment:
One change I had to make recently for `SliverMultiBoxAdaptor` is the ability to check whether the child was actually being kept alive.  https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart#L133

I think it would make sense to add `bool get keptAlive;` to this class and make classes that mix it implement it.  It's used elsewhere in the framework to determine whether elements actually need to be created or inflated.